lxml.h

```diff
@@ -193,12 +234,26 @@ XMLNodeList* XMLNode_children(XMLNode* parent, const char* tag)
 
 char* XMLNode_attr_val(XMLNode* node, char* key)
```

Comment:
I changed the behaviour of this function here to do a `Deep-Clone` as it was leaking memory on line [373](https://github.com/jonahisadev/littlexml/blob/master/lxml.h#L373). It makes it harder to `free` using your utility functions if the memory is pointed to by both the `desc` variable and the `doc`, a Deep-Clone here ensures that both are independent from one another alleviating this issue. The variable immediately goes out of scope so the overhead of `malloc` duplicated bytes should be sufficiently small.